x-pack/plugins/endpoint/server/routes/resolver/queries/lifecycle.ts

```diff
@@ -0,0 +1,64 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { ResolverQuery } from './base';
+import { JsonObject } from '../../../../../../../src/plugins/kibana_utils/public';
+
+// consider limiting the response size to a reasonable value in case we have a bunch of lifecycle events
+export class LifecycleQuery extends ResolverQuery {
+  protected legacyQuery(endpointID: string, uniquePIDs: string[], index: string): JsonObject {
```

Comment:
❔ If these things have two different signatures and we are using Typescript why separate them into two implementations like this? Why not just one `query` with 
`if index { return legacyStuff } else { return newQueryStuff }`

Reply:
One benefit to having 2 functions is it makes it explicit that classes that implement ResolverQuery are expected to supply queries for both Legacy (phase 0) and phase 1 formats.

Reply:
Also:

1. it makes things substantially easier to read than a bunch of json blobs inlined in conditional statements
2. it gives us an opportunity to wrap a lot of the shared functionality with regard to conditional checking in an abstraction so we can hide a bit of the complexity and don't have to duplicate it everywhere